### PR TITLE
Add missing features for DOMRect API

### PR DESCRIPTION
**Pull request by vinyldarkscratch**
This PR is a part of a project to add missing interfaces and interface features to BCD that are from an active spec (including WICG specs) and is supported in at least one browser.  This particular PR adds the missing features of the DOMRect API, populating the results using data from the [mdn-bcd-collector](https://mdn-bcd-collector.appspot.com) project (v3.2.1).

Tests Used: https://mdn-bcd-collector.appspot.com/tests/api/DOMRect

### CODE REVIEW

File: api/DOMRect.json [hunk 1]
```diff
@@ -225,6 +225,106 @@
           }
         }
       },
+      "height": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/DOMRectReadOnly/height",
```
Comment (foolip) [inline api/DOMRect.json:230]: Can you remove "readonly" throughout? Let's omit `mdn_url` if it would be 404.

**Review by foolip — CHANGES_REQUESTED** [review]

File: api/DOMRect.json [hunk 2]
```diff
@@ -225,6 +225,104 @@
           }
         }
       },
+      "height": {
+        "__compat": {
+          "spec_url": "https://drafts.fxtf.org/geometry/#dom-domrectreadonly-height",
```
Comment (foolip) [inline api/DOMRect.json:230]: @vinyldarkscratch this still has "readonly" in the URL. This file shouldn't link to those :)

**Review by foolip — CHANGES_REQUESTED** [review]

File: api/DOMRect.json (hunk 2 above)
Comment (foolip) [reply api/DOMRect.json:230]: I will make the suggestions.

File: api/DOMRect.json [hunk 3]
```diff
@@ -225,6 +225,104 @@
...
+          "status": {
+            "experimental": false,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },
+      "width": {
+        "__compat": {
+          "spec_url": "https://drafts.fxtf.org/geometry/#dom-domrectreadonly-width",
```
Comment (foolip) [inline api/DOMRect.json:279]:
```suggestion
          "spec_url": "https://drafts.fxtf.org/geometry/#dom-domrect-width",
```

File: api/DOMRect.json (hunk 2 above)
Comment (foolip) [reply api/DOMRect.json:230]:
```suggestion
          "spec_url": "https://drafts.fxtf.org/geometry/#dom-domrect-height",
```

File: api/DOMRect.json [hunk 4]
```diff
@@ -272,6 +370,102 @@
             "deprecated": false
           }
         }
+      },
+      "x": {
+        "__compat": {
+          "spec_url": "https://drafts.fxtf.org/geometry/#dom-domrectreadonly-x",
```
Comment (foolip) [inline api/DOMRect.json:376]:
```suggestion
          "spec_url": "https://drafts.fxtf.org/geometry/#dom-domrect-x",
```

File: api/DOMRect.json [hunk 5]
```diff
@@ -272,6 +370,102 @@
...
+          "status": {
+            "experimental": false,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },
+      "y": {
+        "__compat": {
+          "spec_url": "https://drafts.fxtf.org/geometry/#dom-domrectreadonly-y",
```
Comment (foolip) [inline api/DOMRect.json:424]:
```suggestion
          "spec_url": "https://drafts.fxtf.org/geometry/#dom-domrect-y",
```

File: api/DOMRect.json [hunk 6]
```diff
@@ -225,6 +225,104 @@
           }
         }
       },
+      "height": {
+        "__compat": {
+          "spec_url": "https://drafts.fxtf.org/geometry/#dom-domrectreadonly-height",
+          "support": {
+            "chrome": {
+              "version_added": "61"
```
Comment (foolip) [inline api/DOMRect.json:233]:
```suggestion
              "version_added": "2"
```

For the properties that were on `ClientRect`, I think we should mark support as going back that far. That's what we do with other prefixed and alt.name. entries.

File: api/DOMRect.json [hunk 7]
```diff
@@ -225,6 +225,104 @@
...
+              "version_added": "79"
+            },
+            "firefox": {
+              "version_added": "27"
+            },
+            "firefox_android": {
+              "version_added": "27"
+            },
+            "ie": {
+              "version_added": false,
```
Comment (foolip) [inline api/DOMRect.json:248]:
```suggestion
              "version_added": "4",
```

And no note needed. These were copied from `DOMRectReadOnly.json` but I think we should remove everything about `ClientRect` from there, including the notes.

**Review by foolip — CHANGES_REQUESTED** [review]

File: api/DOMRect.json (hunk 7 above)
Comment (foolip) [reply api/DOMRect.json:248]: I've sent https://github.com/mdn/browser-compat-data/pull/11694 for that.

File: api/DOMRect.json [hunk 8]
```diff
@@ -225,6 +225,102 @@
...
+              "version_added": "≤12.1"
+            },
+            "safari": {
+              "version_added": "4"
+            },
+            "safari_ios": {
+              "version_added": "3.2"
+            },
+            "samsunginternet_android": {
+              "version_added": "8.0"
```
Comment (foolip) [inline api/DOMRect.json:263]:
```suggestion
              "version_added": "1.0"
```

File: api/DOMRect.json [hunk 9]
```diff
@@ -225,6 +225,102 @@
...
+            },
+            "webview_android": {
+              "version_added": "≤37"
```
Comment (foolip) [inline api/DOMRect.json:266]:
```suggestion
              "version_added": "2"
```

File: api/DOMRect.json [hunk 10]
```diff
@@ -225,6 +225,102 @@
...
+              "version_added": "3"
+            },
+            "firefox_android": {
+              "version_added": "4"
+            },
+            "ie": {
+              "version_added": "4"
+            },
+            "opera": {
+              "version_added": "≤12.1"
```
Comment (foolip) [inline api/DOMRect.json:251]:
```suggestion
              "version_added": "9.5"
```

File: api/DOMRect.json [hunk 11]
```diff
@@ -225,6 +225,102 @@
...
+            },
+            "opera_android": {
+              "version_added": "≤12.1"
```
Comment (foolip) [inline api/DOMRect.json:254]:
```suggestion
              "version_added": "10.1"
```

File: api/DOMRect.json [hunk 12]
```diff
@@ -225,6 +225,102 @@
...
+              "version_added": "3"
+            },
+            "firefox_android": {
+              "version_added": "4"
+            },
+            "ie": {
+              "version_added": "4"
+            },
+            "opera": {
+              "version_added": "≤12.1"
```
Comment (foolip) [inline api/DOMRect.json:299]:
```suggestion
              "version_added": "9.5"
```

File: api/DOMRect.json [hunk 13]
```diff
@@ -225,6 +225,102 @@
...
+            },
+            "opera_android": {
+              "version_added": "≤12.1"
```
Comment (foolip) [inline api/DOMRect.json:302]:
```suggestion
              "version_added": "10.1"
```

File: api/DOMRect.json [hunk 14]
```diff
@@ -225,6 +225,102 @@
...
+            },
+            "safari": {
+              "version_added": "4"
+            },
+            "safari_ios": {
+              "version_added": "3.2"
+            },
+            "samsunginternet_android": {
+              "version_added": "8.0"
```
Comment (foolip) [inline api/DOMRect.json:311]:
```suggestion
              "version_added": "1.0"
```

File: api/DOMRect.json [hunk 15]
```diff
@@ -225,6 +225,102 @@
...
+            },
+            "webview_android": {
+              "version_added": "≤37"
```
Comment (foolip) [inline api/DOMRect.json:314]:
```suggestion
              "version_added": "2"
```

**Comment by foolip** [comment]
One important fact when reviewing this is to know what properties `ClientRect` originally had. I've tested in IE 11, Safari 4 and Firefox 10 and all have `top`, `right`, `bottom`, `left`, `width` and `height`. I'll assume this was universally true for all `ClientRect` implementations.

File: api/DOMRect.json [hunk 16]
```diff
@@ -272,6 +368,102 @@
             "deprecated": false
           }
         }
+      },
+      "x": {
```
Comment (foolip) [inline api/DOMRect.json:372]: This is `x` which wasn't part of `ClientRect`. Can you make this and `y` match the parent feature instead?

**Review by foolip — CHANGES_REQUESTED** [review]
I noticed that Samsung Internet 8.0 was probably wrong, and then more :)

There's a problem with treating `ClientRect` as an alias I didn't realize before, I'll write a separate comment.

**Comment by foolip** [comment]
Trying to make the data consistently represent the earlier support of `ClientRect` runs into a problem. As mentioned above, `ClientRect` had 6 properties. But only `width` and `height` are on `DOMRect` (and `DOMRectReadOnly`), the other 4 (`top`, `right`, `bottom`, `left`) are on `DOMRectReadOnly` only.

This means that we'd need to treat `ClientRect` as an old alias of *both* `DOMRect` and `DOMRectReadOnly`, but I just removed the latter in https://github.com/mdn/browser-compat-data/pull/11694.

It's tempting to say we should just forget about `ClientRect` and `ClientRectList`, but doing that will make BCD inconsistent, seemingly claiming support of APIs like `getBoundingClientRect()` before its return type is supported.

@ddbeck @Elchi3 do you have thoughts on how to handle this?

**Comment by ddbeck** [comment]
Hmm. My initial reaction is to favor consistency, even though it's a bit of a hassle to reinstate what you just removed. This isn't even so much for the benefit of being outwardly consistent as it is for not doing surprising things that we'll stumble upon later.

**Comment by foolip** [comment]
> Hmm. My initial reaction is to favor consistency, even though it's a bit of a hassle to reinstate what you just removed. This isn't even so much for the benefit of being outwardly consistent as it is for not doing surprising things that we'll stumble upon later.

It was a bit of a hassle, but I've sent https://github.com/mdn/browser-compat-data/pull/11960 now. I think having the alias in both places is the right call in the end.

File: api/DOMRect.json (hunk 16 above)
Comment (foolip) [reply api/DOMRect.json:372]: I'll suggest the specific edits.

File: api/DOMRect.json [hunk 17]
```diff
@@ -272,6 +368,102 @@
...
+        "__compat": {
+          "spec_url": "https://drafts.fxtf.org/geometry/#dom-domrect-x",
+          "support": {
+            "chrome": {
+              "version_added": "2"
```
Comment (foolip) [inline api/DOMRect.json:377]:
```suggestion
              "version_added": "61"
```

File: api/DOMRect.json [hunk 18]
```diff
@@ -272,6 +368,102 @@
...
+            },
+            "chrome_android": {
+              "version_added": "18"
```
Comment (foolip) [inline api/DOMRect.json:380]:
```suggestion
              "version_added": "61"
```

File: api/DOMRect.json [hunk 19]
```diff
@@ -272,6 +368,102 @@
...
+            },
+            "edge": {
+              "version_added": "12"
```
Comment (foolip) [inline api/DOMRect.json:383]:
```suggestion
              "version_added": "79"
```

File: api/DOMRect.json [hunk 20]
```diff
@@ -272,6 +368,102 @@
...
+            },
+            "firefox": {
+              "version_added": "31"
+            },
+            "firefox_android": {
+              "version_added": "31"
+            },
+            "ie": {
+              "version_added": "4"
```
Comment (foolip) [inline api/DOMRect.json:392]:
```suggestion
              "version_added": false
```

File: api/DOMRect.json [hunk 21]
```diff
@@ -272,6 +368,102 @@
...
+            },
+            "opera": {
+              "version_added": "≤12.1"
```
Comment (foolip) [inline api/DOMRect.json:395]:
```suggestion
              "version_added": "48"
```

File: api/DOMRect.json [hunk 22]
```diff
@@ -272,6 +368,102 @@
...
+            },
+            "opera_android": {
+              "version_added": "≤12.1"
```
Comment (foolip) [inline api/DOMRect.json:398]:
```suggestion
              "version_added": "45"
```

File: api/DOMRect.json [hunk 23]
```diff
@@ -272,6 +368,102 @@
...
+            },
+            "safari": {
+              "version_added": "4"
```
Comment (foolip) [inline api/DOMRect.json:401]:
```suggestion
              "version_added": "10.1"
```

File: api/DOMRect.json [hunk 24]
```diff
@@ -272,6 +368,102 @@
...
+            },
+            "safari_ios": {
+              "version_added": "3.2"
```
Comment (foolip) [inline api/DOMRect.json:404]:
```suggestion
              "version_added": "10.3"
```

File: api/DOMRect.json [hunk 25]
```diff
@@ -272,6 +368,102 @@
...
+            },
+            "samsunginternet_android": {
+              "version_added": "8.0"
+            },
+            "webview_android": {
+              "version_added": "≤37"
```
Comment (foolip) [inline api/DOMRect.json:410]:
```suggestion
              "version_added": "61"
```

File: api/DOMRect.json [hunk 26]
```diff
@@ -272,6 +368,102 @@
...
+            "deprecated": false
+          }
+        }
+      },
+      "y": {
+        "__compat": {
+          "spec_url": "https://drafts.fxtf.org/geometry/#dom-domrect-y",
+          "support": {
+            "chrome": {
+              "version_added": "2"
```
Comment (foolip) [inline api/DOMRect.json:425]:
```suggestion
              "version_added": "61"
```

File: api/DOMRect.json [hunk 27]
```diff
@@ -272,6 +368,102 @@
...
+            },
+            "chrome_android": {
+              "version_added": "18"
```
Comment (foolip) [inline api/DOMRect.json:428]:
```suggestion
              "version_added": "61"
```

File: api/DOMRect.json [hunk 28]
```diff
@@ -272,6 +368,102 @@
...
+            },
+            "edge": {
+              "version_added": "12"
```
Comment (foolip) [inline api/DOMRect.json:431]:
```suggestion
              "version_added": "79"
```

File: api/DOMRect.json [hunk 29]
```diff
@@ -272,6 +368,102 @@
...
+            },
+            "firefox": {
+              "version_added": "31"
+            },
+            "firefox_android": {
+              "version_added": "31"
+            },
+            "ie": {
+              "version_added": "4"
```
Comment (foolip) [inline api/DOMRect.json:440]:
```suggestion
              "version_added": false
```

File: api/DOMRect.json [hunk 30]
```diff
@@ -272,6 +368,102 @@
...
+            },
+            "opera": {
+              "version_added": "≤12.1"
```
Comment (foolip) [inline api/DOMRect.json:443]:
```suggestion
              "version_added": "48"
```

File: api/DOMRect.json [hunk 31]
```diff
@@ -272,6 +368,102 @@
...
+            },
+            "opera_android": {
+              "version_added": "≤12.1"
```
Comment (foolip) [inline api/DOMRect.json:446]:
```suggestion
              "version_added": "45"
```

File: api/DOMRect.json [hunk 32]
```diff
@@ -272,6 +368,102 @@
...
+            },
+            "safari": {
+              "version_added": "4"
```
Comment (foolip) [inline api/DOMRect.json:449]:
```suggestion
              "version_added": "10.1"
```

File: api/DOMRect.json [hunk 33]
```diff
@@ -272,6 +368,102 @@
...
+            },
+            "safari_ios": {
+              "version_added": "3.2"
```
Comment (foolip) [inline api/DOMRect.json:452]:
```suggestion
              "version_added": "10.3"
```

File: api/DOMRect.json [hunk 34]
```diff
@@ -272,6 +368,102 @@
...
+            },
+            "samsunginternet_android": {
+              "version_added": "8.0"
+            },
+            "webview_android": {
+              "version_added": "≤37"
```
Comment (foolip) [inline api/DOMRect.json:458]:
```suggestion
              "version_added": "61"
```

**Review by foolip — APPROVED** [review]

**Pull request merged**